cmd/clusterctl/cmd/init_list_images.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+)
+
+var initListImagesCmd = &cobra.Command{
+	Use:   "list-images",
+	Short: "Lists the container images required for initializing the management cluster",
+	Long: LongDesc(`
+		Lists the container images required for initializing the management cluster.
+
+		See https://cluster-api.sigs.k8s.io for more details.`),
+
+	Example: Examples(`
+		# Lists the container images required for initializing the management cluster.
+		#
+		# Note: This command is a dry-run; it won't perform any action other than printing to screen.
+		clusterctl init list-images --infrastructure aws`),
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runInitListImages()
+	},
+}
+
+func runInitListImages() error {
+	c, err := client.New(cfgFile)
+	if err != nil {
+		return err
+	}
+
+	options := client.InitOptions{
+		Kubeconfig:              client.Kubeconfig{Path: initOpts.kubeconfig, Context: initOpts.kubeconfigContext},
```

Comment:
This command should also support listing the images of the CoreProvider. We should support `--core`. 

The user should be able to do `clusterctl init list-images --core cluster-api:v1.2.0` to be able to list the images used by the core provider.